frontend/app/src/pages/gameVisu/GameVisuPage.tsx

```diff
@@ -0,0 +1,208 @@
+import HexagonalBoard from "@/components/hexagonalBoard/HexagonalBoard.tsx";
+import "./gameVisu.css";
+import Header from "@/components/header/Header.tsx";
+import Footer from "@/components/footer/Footer.tsx";
+import { useEffect, useRef, useState } from "react";
+import { HexagonalTileProps } from "@/components/hexagonalTile/HexagonalTile.tsx";
+import { LobbyStatus } from "@/domain/LobbyStatus.ts";
+import { useLobby } from "@/services/lobbyContext/LobbyContextHelpers.ts";
+import { getLobbyStatusV1 } from "@/services/rest/LobbyRest.ts";
+import { useNavigate } from "react-router-dom";
+import { Separator } from "@/components/ui/separator.tsx";
+
+// Not explicitly tested atm, because I dont have the time for it, this is mostly visual and we are nearing the deadline :(
+function GameVisuPage() {
+	const delay = (ms: number | undefined) =>
+		new Promise((res) => setTimeout(res, ms));
+
+	const [lobbyStatus, setLobbyStatus] = useState<LobbyStatus>();
+	const lobby = useLobby();
+	const navigate = useNavigate();
+	const interval_time: number = 33;
+	const counterRef = useRef(0);
+	const [derivedValue, setDerivedValue] = useState(counterRef.current);
+
+	useEffect(() => {
+		if (lobby) {
+			getLobbyStatusV1(lobby.id).then((status) => {
+				console.log("setting lobby status");
+				setLobbyStatus(status);
+				console.log("DONE setting lobby status");
+			});
+		}
+	}, [lobby]);
+
+	useEffect(() => {
+		const interval = setInterval(async () => {
+			if (lobbyStatus == undefined) {
+				console.log("lobby status undefined");
+				return;
+			}
+			if (counterRef.current >= lobbyStatus.visualizationData.length) {
+				console.log("Done visualizing!");
+				clearInterval(interval);
+				await delay(3000);
+				navigate("/result-display");
+				return;
+			}
+
+			if (boardRef.current == null) {
+				console.log("board ref undefined");
+				return;
+			}
+
+			const previousIteration =
+				counterRef.current >= 1
+					? lobbyStatus.visualizationData[counterRef.current - 1]
+					: null;
+			const currentIteration =
+				lobbyStatus.visualizationData[counterRef.current];
+
+			const playerColor =
+				currentIteration.playerId === "playerA" ? "#FF006E" : "#00FFFF";
+
+			const updates: {
+				row: number;
+				col: number;
+				props: Partial<HexagonalTileProps>;
+			}[] = [];
+
+			// Queue updates instead of applying directly
+			for (const writeIndex of currentIteration.memoryWrites) {
+				updates.push({
+					row: Math.floor((writeIndex % 8192) / 128),
```

Comment:
maybe put these numbers(8192 and 128) into const variables to avoid "magic numbers"